bids/reports/parsing.py

```diff
@@ -384,96 +319,74 @@ def final_paragraph(metadata):
-           This section was (in part) generated
-           automatically using pybids ({meth_vers}).
-           '''.format(software_str=software_str,
-                      meth_vers=__version__)
-    desc = desc.replace('\n', ' ').lstrip()
-    while '  ' in desc:
-        desc = desc.replace('  ', ' ')
-
+        software_str = ""
+    desc = (
+        "Dicoms were converted to NIfTI-1 format{software_str}. "
+        "This section was (in part) generated automatically using pybids "
+        "({meth_vers}).".format(
+            software_str=software_str,
+            meth_vers=__version__,
+        )
+    )
     return desc
 
 
-def parse_niftis(layout, niftis, subj, config, **kwargs):
-    """
-    Loop through niftis in a BIDSLayout and generate the appropriate description
-    type for each scan. Compile all of the descriptions into a list.
+def parse_files(layout, data_files, sub, config):
+    """Loop through files in a BIDSLayout and generate appropriate descriptions.
+
+    Then, compile all of the descriptions into a list.
 
     Parameters
     ----------
     layout : :obj:`bids.layout.BIDSLayout`
         Layout object for a BIDS dataset.
-    niftis : :obj:`list` or :obj:`grabbit.core.File`
+    data_files : :obj:`list` of :obj:`bids.layout.models.BIDSFile`
         List of nifti files in layout corresponding to subject/session combo.
-    subj : :obj:`str`
+    sub : :obj:`str`
         Subject ID.
     config : :obj:`dict`
         Configuration info for methods generation.
     """
-    kwargs = {k: v for k, v in kwargs.items() if v is not None}
+    # Group files into individual runs
+    data_files = collect_associated_files(layout, data_files, extra_entities=["run"])
+
+    # print(data_files)
 
     description_list = []
-    skip_task = {}  # Only report each task once
-    for nifti_struct in niftis:
-        nii_file = nifti_struct.path
-        metadata = layout.get_metadata(nii_file)
-        if not metadata:
-            LOGGER.warning('No json file found for %s', nii_file)
+    # Assume all data have same basic info
+    description_list.append(general_acquisition_info(data_files[0][0].get_metadata()))
+
+    for group in data_files:
+
+        if group[0].entities["datatype"] == "func":
+            group_description = func_info(layout, group, config)
+
+        elif (group[0].entities["datatype"] == "anat") and group[0].entities[
+            "suffix"
+        ].endswith("w"):
+            group_description = anat_info(layout, group, config)
+
+        elif group[0].entities["datatype"] == "dwi":
+            group_description = dwi_info(layout, group, config)
+
+        elif (group[0].entities["datatype"] == "fmap") and group[0].entities[
+            "suffix"
+        ] == "phasediff":
+            group_description = fmap_info(layout, group, config)
+
+        elif group[0].entities["datatype"] in ["eeg", "meg", "beh", "perf"]:
+            warnings.warn(group[0].entities["datatype"] + " not yet supported.")
```

Comment:
FYI added some warnings to catch the case when we run into datatypes or files that are unsupported.

Reply:
That's awesome, thanks!